superset-frontend/src/SqlLab/components/ExploreResultsButton/index.jsx

```diff
@@ -60,57 +52,55 @@ class ExploreResultsButton extends React.PureComponent {
     return [];
   }
 
-  getQueryDuration() {
+  function getQueryDuration() {
     return moment
-      .duration(this.props.query.endDttm - this.props.query.startDttm)
+      .duration(props.query.endDttm - props.query.startDttm)
       .asSeconds();
   }
 
-  getInvalidColumns() {
+  function getInvalidColumns() {
     const re1 = /__\d+$/; // duplicate column name pattern
     const re2 = /^__timestamp/i; // reserved temporal column alias
-
-    return this.props.query.results.selected_columns
+    return props.query.results.selected_columns
       .map(col => col.name)
       .filter(col => re1.test(col) || re2.test(col));
   }
 
-  datasourceName() {
-    const { query } = this.props;
+  function datasourceName() {
     const uniqueId = shortid.generate();
     let datasourceName = uniqueId;
-    if (query) {
-      datasourceName = query.user ? `${query.user}-` : '';
-      datasourceName += `${query.tab}-${uniqueId}`;
+    if (props.query) {
+      datasourceName = props.query.user ? `${props.query.user}-` : '';
+      datasourceName += `${props.query.tab}-${uniqueId}`;
     }
     return datasourceName;
   }
 
-  buildVizOptions() {
-    const { schema, sql, dbId, templateParams } = this.props.query;
+  function buildVizOptions() {
+    const { schema, sql, dbId, templateParams } = props.query;
     return {
       dbId,
       schema,
       sql,
       templateParams,
-      datasourceName: this.datasourceName(),
-      columns: this.getColumns(),
+      datasourceName: datasourceName(),
+      columns: getColumns(),
     };
   }
 
-  renderTimeoutWarning() {
+  function renderTimeoutWarning() {
```

Comment:
Please convert this to an arrow function.

Also - GH actions says that this function is defined but never used. If you come across any unused functions like this, please remove them.